pkg/sink/sink.go

```diff
@@ -93,26 +93,50 @@ func (r Sink) HandleEvent(response http.ResponseWriter, request *http.Request) {
 	eventLog.Debugf("EventListener: %s in Namespace: %s handling event (EventID: %s) with path %s, payload: %s and header: %v",
 		r.EventListenerName, r.EventListenerNamespace, eventID, request.URL.Path, string(event), request.Header)
 	var trItems []*triggersv1.Trigger
```

Comment:
could we extract this bit into its own function (findTriggersToProcess or something)?

Reply:
so @dibyom I was considering this, but I've been thinking about a broader strategy of placing a watch on the eventlistener object and then the target trigger queries, and using a controller-like strategy so the triggers aren't calculated every HTTP request. This is in line with what im starting to lay out in #977, hoping that this strategy will allow us to use watches in the sink process rather than resolving an array each request.

What would you like to do here?

Reply:
Even in that scenario we'd have a call like triggers := getTriggersToProcess() or something right? I'd still extract it out so that the code is a bit more readable for now.

As for actually using a watch --> yeah that would be interesting and probably a bit more efficient. We could create an initial list of triggers on startup and then keep a watch (assuming label/namespace selectors are used). And then simply return that list of triggers.

We're already reading off of a lister cache though so it might not be that much more efficient though the code could be a bit more simpler perhaps?